tests/e2e/test_assign_entities.py

```diff
@@ -384,3 +426,44 @@ def test_command_assign_entities_resource_not_processed(
 
     assert "unprocessed_resource" in caplog.text
     assert "not been processed" in caplog.text
+
+
+def test_assign_entities_unique_assignment(
+    collection_dir, pipeline_dir, specification_dir, organisation_path, mock_resource
```

Comment:
If I'm not mistaken you made a fixture second_mock_resource but didn't actually pass it in here?